emop-maven-plugin/src/main/java/edu/cornell/AffectedSpecsMojo.java

```diff
@@ -34,11 +49,92 @@ public void execute() throws MojoExecutionException {
         MessageHandler mh = new MessageHandler();
         compiler.run(arguments, mh);
         IMessage[] ms = mh.getMessages(IMessage.WEAVEINFO, false);
+        computeMapFromMessage(ms);
+        writeMapToFile("txt");
         Writer.writeToFile(Arrays.asList(ms), getArtifactsDir() + File.separator + "join-points");
         getLog().info("[eMOP] Number of impacted classes: " + getImpacted().size());
         getLog().info("[eMOP] Number of messages to process: " + Arrays.asList(ms).size());
     }
 
+    /**
+     * Compute a mapping from affected classes to specifications based on the messages from AJC.
+     * TODO: Handling output messages are definitely slower than reading them directly from AJC
+     * @param ms An array of IMessage objects
+     */
+    private void computeMapFromMessage(IMessage[] ms) throws MojoExecutionException {
+        for (IMessage message : ms) {
+            String[] lexedMessage = message.getMessage().split("'");
+            String key = lexedMessage[CLASS_IDX_IN_MSG];
+            String value = lexedMessage[SPEC_IDX_IN_MSG].substring(TRIMMED_SPEC_NAME_IDX);
+            if (!classToSpecs.containsKey(key)) {
+                classToSpecs.put(key, new HashSet<>());
+            }
+            classToSpecs.get(key).add(value);
+        }
+    }
+
+    /**
+     * Write map from class to specs in either text or binary format.
+     * @param format Output format of the map, text or binary
+     */
+    private void writeMapToFile(String format) throws MojoExecutionException {
+        if (format.equals("bin")) {
+            // Referenced from https://www.geeksforgeeks.org/how-to-serialize-hashmap-in-java/
+            try {
+                FileOutputStream fos = new FileOutputStream(getArtifactsDir() + File.separator + "classToSpecs.bin");
```

Comment:
use try-with-resources, so that you do not need to close these streams: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html